app/src/main/java/org/oppia/android/app/preview/TopicPreviewViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package org.oppia.android.app.preview
+
+import android.content.Context
+import android.view.View
+import androidx.databinding.ObservableField
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import org.oppia.android.R
+import org.oppia.android.app.fragment.FragmentScope
+import org.oppia.android.app.model.Topic
+import org.oppia.android.app.viewmodel.ObservableViewModel
+import org.oppia.android.util.formattor.FileSizeConversionUtil
+import org.oppia.android.util.parser.html.TopicHtmlParserEntityType
+import javax.inject.Inject
+
+/** [ViewModel] for showing topic preview details. */
+@FragmentScope
+class TopicPreviewViewModel @Inject constructor(
+  private val context: Context,
+  fragment: Fragment,
+  @TopicHtmlParserEntityType val entityType: String
+) : ObservableViewModel() {
+
+  private val topicPreviewListener = fragment as TopicPreviewListener
+
+  /** Topic observable instance. */
+  val topic = ObservableField<Topic>(Topic.getDefaultInstance())
+  /** Size of the topic. */
+  val topicSize = ObservableField<String>("")
```

Comment:
Needs a unit--should it be bytes?

Also, the size is presumably an Int, not a String.